src/types.ts

```diff
@@ -0,0 +1,8 @@
+
+export interface XkcdComicProps {
+    showNav?: boolean;
+    showExplain?: boolean;
+    comicNumber?: number;
+    useProxy?: boolean;
```

Comment:
You don't actually need the useProxy and proxyUrls. Instead you could try to get the proxy setting from config something like this:
```ts
const proxies = config.getOptionalConfig('proxy');
const useProxy = proxies?.has('/xkcd-proxy');
const proxyUrl = proxies?.getOptionalConfig('/xkcd-proxy')?.getOptionalString('target');
```
Or maybe if this works:
```ts
const proxyUrl = config.getOptionalString('proxy./xkcd-proxy.target');
const useProxy = Boolean(proxyUrl);
```

Reply:
I mentioned in the README how to setup proxy in the app config.
I checked the code and reading of the proxy settings failed (for both `config.getOptionalConfig('proxy')` and `config.has('proxy')`). I'd like to use it to show some more specific error in case the admin forgets to add to specify the proxy settings. Maybe these settings are not available from some reason. Eg. `config.getString('backend.baseUrl')` works properly. So far I think it's a common practice to use and require proxy settings, so it's OK.